client/my-sites/plugins/disconnect-jetpack/disconnect-jetpack-dialog.jsx

```diff
@@ -39,14 +39,15 @@ module.exports = React.createClass( {
 	},
 
 	disconnectJetpack: function() {
-		var jetpackSites = sites.getJetpack();
+		var jetpackSites = sites.getJetpack(),
+			selectedSite = sites.getSelectedSite();
 
 		// remove any error and completed notices
 		SitesListActions.removeSitesNotices( [ { status: 'error' }, { status: 'completed' } ] );
 
 		if ( this.props.site ) {
 			SitesListActions.disconnect( this.props.site );
-			if ( this.props.redirect && jetpackSites.some( this.jetpackSiteRemain ) ) {
+			if ( selectedSite === this.props.site && this.props.redirect && jetpackSites.some( this.jetpackSiteRemain ) ) {
```

Comment:
I think we can, let's wait for @enejb approval, maybe he knows something we don't :)